### PR TITLE
Solve wording caching if multiple/no addresses

### LINKED ISSUE #28983: FO - Bad wording on footer with "Add a first address" and "Addresses"
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

To have the good wording for Addresses on "Your account" block on footer, we need to clear cache on the BO after every action about Addresses part on customer account

Two cases of the bug

1. Create a new account without any addresses, the "Add a first address" wording is well displayed. If we add an address to this account, the wording doesn't change:

https://user-images.githubusercontent.com/2854834/177586591-7fd16ebc-1cf4-4695-928e-877475307b63.mp4

2. Create a new account without any addresses, , the "Add a first address" wording is well displayed. Sign out and connect with an existing customer account with registered addresses, the wording doesn't change:

https://user-images.githubusercontent.com/2854834/177586683-a480a5e0-e91d-402c-ae80-64210c12f458.mp4

Tested and error on 8.0.0 version.
On 1.7.8.6 / 1.7.7.8, this feature is not available.

### Expected behavior

The wording on this block should be modified: 
- no addresses: "Add a first address" must be displayed
- when the first address is added: "Addresses" must be displayed 
- remove all addresses of the customer account: "Add a first address" must be displayed

### Steps to reproduce

**Case 1: create only new account with addresses**
1. Go to FO > Sign in on header > Click on "No account? Create one here" link
2. Fill the form and save
3. Go to footer, "Your account" block > see the good wording "Add a first address"
4. Click on this link > Fill the form with your address and save
5. Go back to the footer > See error, the wording has not changed to "Addresses"
6. Go to BO > Advanced Parameters > Performance > Click on "Clear cache" button on the top of the page > Wait the successful message
7. Go back to FO > Refresh the page > The wording on the footer has changed to "Addresses"

**Case 2: create new account with addresses, sign out and sign in with existing account**

1. Go to FO > Sign in on header > Click on "No account? Create one here" link
2. Fill the form and save
3. Go to footer, "Your account" block > see the good wording "Add a first address"
4. Sign out and sign in again with an existing account with addresses already created, for example "pub@prestashop.com"
5. Go back to the footer > See error, the wording is still "Add a first address" instead of "Addresses"
6. Go to BO > Advanced Parameters > Performance > Click on "Clear cache" button on the top of the page > Wait the successful message
7. Go back to FO > Refresh the page > The wording on the footer has changed to "Addresses"

### PrestaShop version(s) where the bug happened

8.0.0

### PHP version(s) where the bug happened

7.3

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by LouisAUTHIE**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Fixes wording problem switching from singular to plural on addresses by clearing cache on addresses updating.
| Type?         | bug fix
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/28983
| How to test?  | Use the test protocol in the solved issue (add and remove addresses)

### CODE REVIEW

File: ps_customeraccountlinks.php [hunk 1]
```diff
@@ -86,11 +86,15 @@ public function _clearCache($template, $cache_id = null, $compile_id = null)
 
     public function renderWidget($hookName = null, array $configuration = [])
     {
-        if (!$this->isCached($this->templateFile, $this->getCacheId('ps_customeraccountlinks'))) {
+        $tpl_key = 'ps_customeraccountlinks';
+        if(Customer::getAddressesTotalById($this->context->customer->id)){
+            $tpl_key.="_multiple";
+        }
```
Comment (ghost) [inline ps_customeraccountlinks.php]: We can use an existant variable : 

```
        $tpl_key = 'ps_customeraccountlinks';
        if($this->context->smarty->tpl_vars['customer']->value['addresses']){
            $tpl_key.="_multiple";
        }
```

**Comment by LouisAUTHIE** [comment]
Ok for me for this good idea tkx

**Comment by matks** [comment]
@LouisAUTHIE Apologies for late answer. Thank you for the PR I added a little [commit](https://github.com/PrestaShop/ps_customeraccountlinks/pull/41/commits/359891a60f54307c6bbf56f5402b6e2f555f0972) for codestyle and I approve the PR that can be tested by QA.

**Review by matks — APPROVED** [review]

**Review by sLorenzini — APPROVED** [review]
hello @LouisAUTHIE,

Thank you for this PR.
Tested the module PR on 8.0.0 version only as written on the fixed ticket, php version 7.3.

Addresses footer wording is good on all cases:

- Case 1 from the ticket: 

https://user-images.githubusercontent.com/2854834/189148048-7444eca9-3d47-4206-a692-9ec26b5e591b.mp4

- Case 2 from the ticket

https://user-images.githubusercontent.com/2854834/189148089-89f08800-eed0-4526-ac58-6f3575e7ae9d.mp4

- Another case (delete and re-add an address with an account):

https://user-images.githubusercontent.com/2854834/189148117-f81d0f0d-0bec-481f-b931-5edf1e1eb7ee.mp4

QA approved ✅ 

Thanks

**Comment by matks** [comment]
Thank you @sLorenzini and @LouisAUTHIE 

Great news to see this PR finally merged 😁

**Comment by LouisAUTHIE** [comment]
Thx all

**Pull request merged**